internal/api/v1/service/validate.go

```diff
@@ -61,7 +60,7 @@ func ValidateService(
 	srv, err := client.GetRelease(releaseName)
 	if err != nil {
 		if errors.Is(err, helmdriver.ErrReleaseNotFound) {
-			return apierror.NewNotFoundError(fmt.Sprintf("%s - %s", err.Error(), releaseName))
+			return apierror.NewNotFoundError(errors.Wrapf(err, "release %s not found", releaseName))
```

Comment:
I can see this being more sensible than what we had, i.e. providing the entire base error instead of just its message.
OTOH the 

```
-		return apierror.NewNotFoundError("service not found")
+		return apierror.NewNotFoundError(errors.New("service not found"))
```

in line 47 above just becomes more verbose than necessary.

How about having two functions, with slightly differing signatures, i.e. the old function, keeping its signature

> NewXError (msg string, ...)

and a new

> NewXFromError (err error, ...)

with the new signature for where we wish to pass in an underlying error ?

And coming back to this place, as written even that is actually to verbose, as it does not use the `...string` argument of the old and new function, while it could have:
```suggestion
			return apierror.NewNotFoundError(err, "release", releaseName, "not found")
```

Although that runs afoul of the `", "` separator used in the underlying `strings.Join`. A `" "` separator would be nicer IMHO.

Looking at other places using `fmt.Errorf`, `fmt.Sprintf`, and the like a simple reorg of the error message could get rid of many of these I believe. I.e. instead of `fmt.Errorf("Foo '%s' does not exist", X)` use `"Foo does not exist:", X`, i.e. the dynamic part is just another detail argument of the `NewX...` function.

Reply:
Ah, I've commented before have read this comment. :smile: 
I'm fine with having the two signatures! :+1:

Reply:
Note that `NewX` functions seem to use only the `err.Error()` of the passed error. The `errors.Wrapf` in the callers effectively devolve into an extended error message. and that is something we already can have without the additional Wrap, through the `details` argument.